src/fetch/contracts.py

```diff
@@ -98,12 +101,47 @@ def results(self, load_from: NetworkFile) -> dict[str, bool]:
             partition = addresses[index:index + self.max_batch_size]
             results |= self._limited_is_contract(partition)
 
-        confirmed_contracts = sorted([k for k, v in results.items() if v])
+        confirmed_contracts = sorted([Account(k) for k, v in results.items() if v])
         print(f"found {len(confirmed_contracts)} contracts, writing to file")
-        with open(load_file.filename(), 'w', encoding='utf-8') as file:
-            file.write('\n'.join(confirmed_contracts))
+        write_to_csv(data_list=confirmed_contracts, outfile=load_file)
         return results
 
+    def _limited_balances(self, addresses: list[str]) -> dict[str, int]:
+        request_data = []
+        for index, eth_address in enumerate(addresses):
+            request_data.append({
+                "jsonrpc": "2.0",
+                "method": "eth_getBalance",
+                "params": [eth_address, "latest"],
+                "id": index
+            })
+        response = requests.post(self.node_url, json=request_data)
+        results = {}
+        for result_dict in response.json():
+            try:
+                account = addresses[result_dict['id']]
+                results[account] = int(result_dict['result'], base=16)
+            except KeyError as err:
+                raise IOError(
+                    f"Request for address \"{addresses[result_dict['id']]}\" "
+                    f"failed with response {result_dict}"
+                ) from err
+        return results
+
+    def get_null_balances(self) -> set[str]:
+        print(f"Fetching balances at {len(self.addresses)} addresses on "
+              f"{self.network} (this may take a while)...")
+        results = {}
+        for index in range(0, len(self.addresses), self.max_batch_size):
+            partition = self.addresses[index:index + self.max_batch_size]
+            results |= self._limited_balances(partition)
+
+        null_balances = sorted([Account(k) for k, v in results.items() if v == 0])
+        print(f"found {len(null_balances)} accounts will zero balance, writing to file")
```

Comment:
nanonit: typo will -> with

Reply:
Will fix this in the allocation receipt PR.